### ISSUE TITLE
conventional-commits-parser: commit subject starting with # are ignored

### ISSUE THREAD

**Issue by pvdlg** [conventional-changelog/conventional-changelog#251]
Following the merge of #231, commit message cannot start with `#` anymore.

For example with the following configuration:
```js
parserOpts: {
  headerPattern: /^#(.*?) (.*)$/,
  headerCorrespondence: ['type', 'message']
}
```
and the commit message: `#Fix Message` the parsed commit object does't have the `type` and `message` property.

This works:
```js
parserOpts: {
  headerPattern: /^%(.*?) (.*)$/,
  headerCorrespondence: ['type', 'message']
}
```
With commit `%Fix Message`.

I would suggest that the comments support added in #231 would be considered only in the body of the commit and not in the subject, as I don't think it would make sense to comment the subject.

**Comment by marionebl** [comment]
## TL;DR

* `git commit -m` does not parse lines with leading `#` as comments
* `git commit` does
* There is no way for `conventional-commits-parser` to know if a string originates from `git commit` or `git commit -m`
* I'd revert the changes introducing comment support and document that consumers reading directly from `.git/COMMIT_EDITMSG` (e.g. commitlint) have to filter comments on their own

## Test cases

### git commit -m
```
git commit -m "another thing" -m "# fix: simplify startup" -m "# foo"
git log

commit 93faeca894769bf3bb3a8b65bbd7b1d4a1ccbd70 (HEAD -> master)
Author: Mario Nebl <hello@mario-nebl.de>
Date:   Tue Nov 28 10:24:37 2017 +0100

    another thing

    # fix: simplify startup

    # foo
```

### git commit 
```
git commit
```
```
### in vim

another thing

# fix: simplify startup

# foo

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch master
# Changes to be committed:
#       deleted:    foo
#
```

```
commit 2a5c8ac706c643947274732eda17b10579ff7eef (HEAD -> master)
Author: Mario Nebl <hello@mario-nebl.de>
Date:   Tue Nov 28 10:27:45 2017 +0100

    another thing
```

**Comment by stevemao** [comment]
So, when you do `git commit`, comments are auto stripped. Meaning when you do `git log` you don't know if there was a comment or not. Is that correct? So the comments are already handled by `git commit` and we shouldn't worry about it?

**Comment by marionebl** [comment]
What I am doing with commitlint is reading the raw contents of `.git/COMMIT_EDIT_MSG` and feeding that into the parser. 

I'd argue those qualify as commit message, so having the possibility to support that  usecase optionally would be nice.

In any case I stumbled over both the old (no comment handling) and the current, faulty behaviour (always handling) and would like to centralise the additional complexity of the solution for this at conventional-changelog.

**Comment by marionebl** [comment]
Addendum: The contents of commit message edit file contain the comments.

**Comment by stevemao** [comment]
This is interesting... Go ahead and do it 😄

**Comment by marionebl** [comment]
Already did over at #252. A review would be awesome!
